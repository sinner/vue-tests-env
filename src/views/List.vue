<template>
  <div class="list">
    <ul>
      <StrongRedListItem
        :key="index"
        v-for="(movie, index) in marvelMovies"
        :movie="movie"
      />
    </ul>
  </div>
</template>

<script>
import StrongRedListItem from '@/components/StrongRedListItem.vue';

export default {
  components: {
    StrongRedListItem,
  },

  data() {
    return {
      marvelMovies: [
        'Iron Man',
        'Avengers',
        'Infitinity War',
      ],
    };
  },

};
</script>

<style scoped>
.list {
  text-align: left;
}
</style>
