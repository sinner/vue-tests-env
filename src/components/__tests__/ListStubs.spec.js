import { shallowMount } from '@vue/test-utils';
import List from '@/views/List.vue';

const ListItemStub = {
  template: '<li>{{ movie }}</li>',
  props: ['movie'],
};

describe('List.vue stubs', () => {
  it('should list element using shallowMount and Stub component', async () => {
    const wrapper = shallowMount(List, {
      stubs: {
        ListItem: ListItemStub,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
