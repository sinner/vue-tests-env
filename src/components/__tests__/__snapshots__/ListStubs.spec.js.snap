// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue stubs should list element using shallowMount and Stub component 1`] = `
<div class="list">
  <ul>
    <strongredlistitem-stub movie="Iron Man"></strongredlistitem-stub>
    <strongredlistitem-stub movie="Avengers"></strongredlistitem-stub>
    <strongredlistitem-stub movie="Infitinity War"></strongredlistitem-stub>
  </ul>
</div>
`;
